src/main/java/com/maxmind/db/Reader.java

```diff
@@ -352,19 +352,19 @@ int record = this.startNode(bitLength);
             record = this.readNode(buffer, record, bit);
         }
 
-        return new int[]{record, i};
+        return new long[]{record, i};
     }
 
-    int readNode(ByteBuffer buffer, int nodeNumber, int index)
+    int readNode(Buffer buffer, long nodeNumber, int index)
```

Comment:
I think we should return `long` from this function. We will also need to update it replace the static `decodeInteger` with a `decodeLong`. The issue is that for 32-bit nodes, we could overflow.